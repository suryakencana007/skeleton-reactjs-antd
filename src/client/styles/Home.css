.Home {
  text-align: center;
}

.Home-logo {
  animation: Home-logo-spin infinite 20s linear;
  height: 80px;
}

.Home-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.Home-intro {
  font-size: large;
}

.Home-resources {
  list-style: none;
}

.Home-resources > li {
  display: inline-block;
  padding: 1rem;
}

@keyframes Home-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

/* Style for parallax background*/
/* vh = 1% from viewport heigth
 * vw = 1% from viewport width
 */
.fullscreen {
  height: 100vh; 
}

.background-image-holder {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: 1;
  background: #292929;
  background-size: cover !important;
  background-position: 50% 50% !important;
  transition: all 0.3s ease;
  -webkit-transition: all 0.3s ease;
  -moz-transition: all 0.3s ease;
  opacity: 0;
}
.background-image-holder img {
  display: none;
}
.background-multiply .background-image-holder {
  background-color: #47b475 !important;
  background-blend-mode: multiply;
}
.background-image-holder.fadeIn {
  opacity: 1;
}
.parallax > .background-image-holder,
.parallax .slides li > .background-image-holder {
  height: 100vh;
  top: -50vh;
  transition: all 0s ease !important;
  transition: opacity 0.3s ease !important;
  -webkit-transform-style: preserve-3d;
}
.parallax:first-child .slides li > .background-image-holder,
.parallax:first-child .background-image-holder {
  top: 0;
}
.main-container > a:first-child + .parallax .background-image-holder {
  top: 0;
}
@media all and (max-width: 767px) {
  .parallax > .background-image-holder,
  .parallax .slides li > .background-image-holder {
    top: 0 !important;
    transform: none !important;
    -webkit-transform: none !important;
  }
}