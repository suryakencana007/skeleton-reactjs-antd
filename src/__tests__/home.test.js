import React from 'react';
import renderer from 'react-test-renderer';
import MemoryRouter from 'react-router-dom/MemoryRouter';
import { Provider } from 'react-redux';
import configureStore from 'kao-store';

import Home from 'kao-containers/Home';

const preloadedState = window.__PRELOADED_STATE__ || {}; // eslint-disable-line no-underscore-dangle
const store = configureStore(preloadedState);

it('renders Home correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
