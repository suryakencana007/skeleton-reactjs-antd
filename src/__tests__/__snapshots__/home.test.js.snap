// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
<div
  className="Home"
>
  <div
    className="Home-header"
  >
    <img
      alt="logo"
      className="Home-logo"
      src="react.svg"
    />
    <h2>
      Welcome to Razzle
    </h2>
  </div>
  <p
    className="Home-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     or
    <code>
      src/Home.js
    </code>
     and save to reload.
  </p>
  <ul
    className="Home-resources"
  >
    <li>
      <a
        href="https://github.com/jaredpalmer/razzle"
      >
        Docs
      </a>
    </li>
    <li>
      <a
        href="https://github.com/jaredpalmer/razzle/issues"
      >
        Issues
      </a>
    </li>
    <li>
      <a
        href="https://palmer.chat"
      >
        Community Slack
      </a>
    </li>
  </ul>
  <div>
    Hai, Hello World really excetiing ok ok
    <button
      className="sc-bdVaJa igGFgA"
    >
      PUSH
    </button>
  </div>
  <div>
    <p>
      0
    </p>
    <button
      className="sc-bwzfXH hwcCHa"
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      className="sc-bwzfXH czLwVv"
      onClick={[Function]}
    >
      Decrement
    </button>
  </div>
  <a
    href="/counter"
    onClick={[Function]}
  >
     to counter
  </a>
</div>
`;
